pyvista/core/utilities/helpers.py

```diff
@@ -10,13 +12,14 @@
 
 import pyvista
 from pyvista.core import _vtk_core as _vtk
+from pyvista.core.input_validation.validate import validate_array
 
 from . import transformations
 from .fileio import from_meshio, is_meshio_mesh
 
 
 def wrap(
-    dataset: Optional[Union[np.ndarray, _vtk.vtkDataSet, 'Trimesh', 'Mesh']]
+    dataset: Optional[Union[np.ndarray, _vtk.vtkDataSet, 'Trimesh', 'Mesh', Matrix[float]]]
```

Comment:
Could this technically be of type `Matrix[int]` too? I'm not sure.

Reply:
`int` is allowed by type checkers anywhere `float` is used. So I think `int` is implied here, and doing something like `Union[Matrix[float], Matrix[int]]` is not necessary. (see [PEP484](https://peps.python.org/pep-0484/#the-numeric-tower)). 

As an aside, the mypy plugin from #5285 could mess with this a bit, since that plugin maps `float <-> np.floating` and `int <-> np.integer`... so `np.integer` may also be allowed, though I'm not sure.

But, this `float` and `int` interchangeability for type hints has its quirks, since type checkers won't complain if you do `x: Sequence[float] = [1.0, 2]` since both float and int are allowed.

If you really only want floats and no ints, then  it's best to use a `TypeVar` instead, since they can be bound to `float` explicitly which will cause a type checker to complain, e.g:

``` python
FloatType = TypeVar('FloatType', bound=float)
`x: Sequence[FloatType] = [1.0, 2]`  # not allowed
```

A general `NumberType` TypeVar is already part of `_typing_core`, so perhaps you can just use that, i.e. `Matrix[NumberType]`. This is what's used in #5571 for annotating all array-like inputs.